token-lending/program/src/processor.rs

```diff
@@ -578,6 +587,315 @@ fn _deposit_reserve_liquidity<'a>(
+    }
+
+    static FLASH_REPAY_INSTRUCTION: u8 = 18;
+
+    // loop through instructions, looking for an equivalent repay to this borrow
+    let mut i = current_index + 1;
+    loop {
+        // get the next instruction, die if theres no more
+        if let Ok(ixn) = load_instruction_at_checked(i, sysvar_info) {
+            // check if we have a toplevel repay toward the same reserve
+            // if so, confirm the amount, otherwise next instruction
+            // Note: First byte is instruction ID, next 8 bytes are liquidity_amount
+            if ixn.program_id == *program_id
+                && u8::from_le_bytes([ixn.data[0]]) == FLASH_REPAY_INSTRUCTION
+                && ixn.accounts[4].pubkey == *reserve_info.key
+            {
+                if u64::from_le_bytes(ixn.data[1..9].try_into().unwrap()) == liquidity_amount {
+                    break;
+                } else {
+                    msg!(
+                        "Liquidity amount for flash repay doesn't match borrow or can't be parsed"
+                    );
+                    return Err(LendingError::InvalidFlashRepay.into());
+                }
+            } else {
+                i += 1;
+            }
+        } else {
+            msg!("No flash repay found");
+            return Err(LendingError::NoFlashRepayFound.into());
+        }
+    }
+
+    reserve.liquidity.borrow(flash_loan_amount_decimal)?;
+    reserve.borrowing = true;
+    Reserve::pack(reserve, &mut reserve_info.data.borrow_mut())?;
+
+    spl_token_transfer(TokenTransferParams {
+        source: source_liquidity_info.clone(),
+        destination: destination_liquidity_info.clone(),
+        amount: flash_loan_amount,
+        authority: lending_market_authority_info.clone(),
+        authority_signer_seeds,
+        token_program: token_program_id.clone(),
+    })?;
+
+    Ok(())
+}
+
+fn process_flash_repay_reserve_liquidity(
+    program_id: &Pubkey,
+    liquidity_amount: u64,
+    accounts: &[AccountInfo],
+) -> ProgramResult {
+    if liquidity_amount == 0 {
+        msg!("Liquidity amount provided cannot be zero");
+        return Err(LendingError::InvalidAmount.into());
+    }
+
+    let account_info_iter = &mut accounts.iter();
+    let source_liquidity_info = next_account_info(account_info_iter)?;
+    let destination_liquidity_info = next_account_info(account_info_iter)?;
+    let reserve_liquidity_fee_receiver_info = next_account_info(account_info_iter)?;
+    let host_fee_receiver_info = next_account_info(account_info_iter)?;
+    let reserve_info = next_account_info(account_info_iter)?;
+    let lending_market_info = next_account_info(account_info_iter)?;
+    let user_transfer_authority_info = next_account_info(account_info_iter)?;
+    let sysvar_info = next_account_info(account_info_iter)?;
+    let token_program_id = next_account_info(account_info_iter)?;
+
+    // We don't care about the return value here, so just ignore it.
```

Comment:
I think this comment was copied from somewhere else, but I think in this case we technically do care about the return value since it's just a `ProgramResult`. In the other place, it also returned an int or something

Reply:
Nice catch, yeah. The comment doesn't make sense here because there isn't a meaningful return value. I was returning a value here before representing the borrowed amount, but I forgot to remove the comment after we decided it wasn't necessary.